ante-ls/src/main.rs

```diff
@@ -112,51 +145,110 @@ fn rope_range_to_lsp_range(range: std::ops::Range<usize>, rope: &Rope) -> Range
     }
 }
 
-fn parser_error_diagnostic(err: ParseError<'_>) -> (&std::path::Path, std::ops::Range<usize>, String) {
-    match err {
-        ParseError::Fatal(e) => parser_error_diagnostic(*e),
-        ParseError::InRule(rule, loc) => {
-            (loc.filename, loc.start.index..loc.end.index, format!("failed trying to parse a {}", rule))
-        },
-        ParseError::LexerError(e, loc) => (loc.filename, loc.start.index..loc.end.index, e.to_string()),
-        ParseError::Expected(tokens, loc) => {
-            let message = tokens.into_iter().map(|t| format!("\t - {t}")).collect::<Vec<_>>().join("\n");
-            (loc.filename, loc.start.index..loc.end.index, format!("expected one of:\n {}", message))
-        },
-    }
-}
-
 impl Backend {
     async fn update_diagnostics(&self, uri: Url, rope: &Rope) {
+        let root = match current_dir() {
+            Ok(root) => root,
+            Err(_) => {
+                self.client.log_message(MessageType::ERROR, "Failed to get current directory".to_string()).await;
+                return;
+            },
+        };
+        // We want the filename to be relative to the root for nicer error messages.
+        // This could fail on windows when the root is on a different drive than the file.
```

Comment:
I suppose one instance this can happen is when using libraries stored on different drives